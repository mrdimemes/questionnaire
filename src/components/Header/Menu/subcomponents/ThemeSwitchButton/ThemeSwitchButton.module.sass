@use "src/sass/foundation/mixins"
@use "src/sass/foundation/breakpoints"

.body
  &#{&}
    background-color: transparent
    border: 0
    padding: 0
    @include mixins.square(25px)
    @include mixins.unhighlightable
    @media (max-width: breakpoints.$tablet)
      @include mixins.square(20px)

.theme_light
  background-image: url("/assets/images/icons/light-mode-icon.svg")

.theme_dark
  background-image: url("/assets/images/icons/dark-mode-icon.svg")
