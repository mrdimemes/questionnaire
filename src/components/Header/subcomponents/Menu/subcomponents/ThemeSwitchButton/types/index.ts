export type { ThemeSwitchButtonProps } from "./ThemeSwitchButtonProps";