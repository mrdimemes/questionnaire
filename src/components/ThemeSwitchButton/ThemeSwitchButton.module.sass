@use "src/sass/foundation/mixins"
@use "src/sass/foundation/breakpoints"

.body
  background-color: transparent
  border: 0
  padding: 0
  @include mixins.square(25px)
  @media (max-width: breakpoints.$tablet)
    @include mixins.square(20px)

.icon
  @include mixins.containble
