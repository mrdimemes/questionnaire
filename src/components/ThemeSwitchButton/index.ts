export { default as ThemeSwitchButton } from "./ThemeSwitchButton";