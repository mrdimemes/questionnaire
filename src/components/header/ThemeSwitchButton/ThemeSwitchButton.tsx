import classNames from "classnames";
import { useThemeSelector, useAppDispatch } from "src/redux/hooks";
import { Theme, setTheme, getThemeStyle } from "src/redux/slices/themeSlice";
import { Button } from "src/components/UI/Button";
import styles from "./ThemeSwitchButton.module.sass";

const ThemeSwitchButton = () => {
  const dispatch = useAppDispatch();
  const currentTheme = useThemeSelector();

  const toggleTheme = () => {
    if (currentTheme === Theme.Light) {
      dispatch(setTheme(Theme.Dark))
    } else {
      dispatch(setTheme(Theme.Light))
    }
  }

  return (
    <Button
      onClick={toggleTheme}
      className={classNames(
        styles.body,
        getThemeStyle(styles, currentTheme)
      )}
    />
  )
}

export default ThemeSwitchButton;