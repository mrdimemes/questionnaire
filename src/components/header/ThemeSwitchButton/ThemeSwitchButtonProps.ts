export type ThemeSwitchButtonProps = {
  className?: string,
};