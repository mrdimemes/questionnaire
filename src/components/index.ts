export { App } from "./App";

export { Header } from "./header";

export { PaginationBar } from "./pagination";

export { TagComponent } from "./tags";
export { TagGallery } from "./tags";
export { TagBar } from "./tags";
export { AddTagWidget } from "./tags";

export { QuestionnaireGallery } from "./questionnaires/gallery";
export { QuestionnaireComponent } from "./questionnaires/questionnaire";
export { QuestionnaireEditor } from "./questionnaires/edit";

export { Button } from "./UI/Button";
export { Input } from "./UI/inputs";
export { CheckboxInput } from "./UI/inputs";
export { RadioInput } from "./UI/inputs";
export { TextInput } from "./UI/inputs";
export { EmailInput } from "./UI/inputs";
export { PasswordInput } from "./UI/inputs";
export { NameInput } from "./UI/inputs";
export { TextArea } from "./UI/inputs";
export { Select } from "./UI/inputs";

export { AuthForm } from "./AuthForm";

export { LoadingSpinner } from "./LoadingSpinner";