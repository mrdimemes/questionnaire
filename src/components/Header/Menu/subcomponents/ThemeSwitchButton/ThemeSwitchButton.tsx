import classNames from "classnames";
import { useThemeSelector, useAppDispatch } from "src/redux/hooks";
import { Theme, setTheme } from "src/redux/slices/themeSlice";
import { Button } from "src/components";
import { withTheme } from "src/HOCs";

import styles from "./ThemeSwitchButton.module.sass";

import type { ThemeSwitchButtonProps } from "./types";

const ThemeSwitchButton = ({ className }: ThemeSwitchButtonProps) => {
  const dispatch = useAppDispatch();
  const currentTheme = useThemeSelector();

  const toggleTheme = () => {
    if (currentTheme === Theme.Light) {
      dispatch(setTheme(Theme.Dark));
    } else {
      dispatch(setTheme(Theme.Light));
    }
  };

  return (
    <Button
      onClick={toggleTheme}
      className={classNames(styles.body, className)}
    />
  );
};

export default withTheme(ThemeSwitchButton, styles);