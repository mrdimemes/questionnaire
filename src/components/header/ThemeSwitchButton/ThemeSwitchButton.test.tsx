import { screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderWithProviders } from "src/utils/test-utils";
import { setupStore } from "src/redux";
import { Theme } from "src/redux/slices/themeSlice";
import ThemeSwitchButton from "./ThemeSwitchButton";

describe("Header/ThemeSwitchButton component", () => {
  test("ThemeSwitchButton renders", () => {
    renderWithProviders(<ThemeSwitchButton />);
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("ThemeSwitchButton switches the app's theme", () => {
    const store = setupStore({ theme: { theme: Theme.Light } })
    renderWithProviders(<ThemeSwitchButton />, { store });
    userEvent.click(screen.getByRole("button"));
    expect(store.getState().theme.theme).toEqual(Theme.Dark);
  });

  test("ThemeSwitchButton snapshot", () => {
    const { container } = renderWithProviders(<ThemeSwitchButton />);
    expect(container).toMatchSnapshot();
  });
});