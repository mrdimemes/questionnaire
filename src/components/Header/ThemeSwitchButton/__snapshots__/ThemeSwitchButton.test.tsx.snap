// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header/ThemeSwitchButton component ThemeSwitchButton snapshot 1`] = `
<div>
  <button
    class="body theme_light body theme_light"
  />
</div>
`;
